src/vsr/replica.zig

```diff
@@ -1371,6 +1392,30 @@ pub fn ReplicaType(
             assert(!self.do_view_change_quorum);
             self.do_view_change_quorum = true;
 
+            const dvc_latest = DVCQuorum.dvc_latest(self.do_view_change_from_all_replicas);
+            if (dvc_latest.header.op > self.op_checkpoint_trigger()) {
+                // When the cluster is at a checkpoint ahead of the local checkpoint, abdicate
+                // as primary.
+                //
+                // This serves a few purposes:
+                // 1. Availability: We pick a primary to minimize the number of WAL repairs, to
+                //    minimize the likelihood of a repair-deadlock.
+                // 2. Optimization: The cluster does not need to wait for a lagging replicas before
+                //    prepares/commits can resume.
+                // 3. Simplify repair: A new primary never needs to fast-forward to a new checkpoint.
+
+                log.debug("{}: on_do_view_change: lagging primary; forfeiting " ++
```

Comment:
```suggestion
                log.debug("{}: on_do_view_change: lagging primary; abdicating " ++
```

Let's use `abdicate` as glossary term for when a (would be) primary volonturely decides to step down.

Reply:
My thinking was:
- We already use "abdicating" for the case where the primary isn't receiving acks and has paused sending heartbeats. But this case is recoverable — the primary may receive some acks and _stop_ abdicating.
- Whereas "forfeiting" is immediate and irrevocable.

The terms aren't great, but that is the reason for the distinction.